Async.Model/AsyncLoader.cs

```diff
@@ -182,9 +185,23 @@ private IEnumerable<ItemChange<TItem>> PerformUpdatesInsideLock(IEnumerable<Item
             // NOTE: We preserve any items from seq not found in fetchedUpdates, since they must have been Conj'ed
             var changes = seq
                 .FullOuterJoin(fetchedUpdates, i => i, u => u.Item,
-                    (i, u, k) => u ?? new ItemChange<TItem>(ChangeType.Unchanged, i))
+                    (i, u, k) => u ?? new ItemChange<TItem>(ChangeType.Unchanged, i), identityComparer)
                 .ToArray();  // materialize to prevent multiple enumerations of source
 
+            // TODO: What do we do about invalid item changes, such as update or removal of a non-existing item?
```

Comment:
//4. ALT: We could ignore the Itemchange and send an errorlog to the server so the problem is not ignored and bugs can be lured out. This has the advantage of keeping code simple and helps us find eventual bugs. It does however require us to have a bucket to report errors in AND we need to think sincerely about what to log, so that eventual errorcontext becomes clear.